### ISSUE TITLE
why it runs slowly on iOS，can you give me some tips?

### ISSUE THREAD

**Issue by littleshao** [qaprosoft/carina#611]
hei，it runs slowly on iOS，can you give me some tips?

**Comment by vdelendik** [comment]
Hey @littleshao.
Could you please share exact core version you use. In the latest 6.0.5 I reorganized default fluent waiters using by annotation only. Potentially it could lead to EXPLICIT_TIMEOUT waits before each operation aka click, type etc.

So share with me carina-core version you specified in your pom.xml file and ideally debug console log which can be generated running test with below property:
core_log_level=DEBUG

**Comment by littleshao** [comment]
Thanks！！！
`<properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <carina-core.version>1.0-SNAPSHOT</carina-core.version>
        <httpclient.version>4.5.3</httpclient.version>
        <java.version>1.8</java.version>
        <suite>api</suite>
    </properties>`
`#============== WebDirver configuration ==============#
implicit_timeout=0
explicit_timeout=0
retry_interval=5
thread_count=5
data_provider_thread_count=-1
auto_download=false
auto_download_apps=NULL
custom_artifacts_folder=NULL
scroll_to_element_y_offset=120
#=====================================================#
`
and，When using xpath, the execution is slow；it runs more fast when using MobileBy.iOSNsPredicateString；but I want it to run faster；
debug log ，I will provide later.

**Comment by littleshao** [comment]
Thanks!!!

>     <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <carina-core.version>1.0-SNAPSHOT</carina-core.version>
        <httpclient.version>4.5.3</httpclient.version>
        <java.version>1.8</java.version>
        <suite>api</suite>
    </properties>

> #============== WebDirver configuration ==============#
implicit_timeout=0
explicit_timeout=0
retry_interval=5
thread_count=5
data_provider_thread_count=-1
auto_download=false
auto_download_apps=NULL
custom_artifacts_folder=NULL
scroll_to_element_y_offset=120
#=====================================================#


and it runs slowly when using xpath,but it be more faster when using MobileBy.iOSNsPredicateString;
I want it to run faster.
debug log ,I will provide later.

**Comment by kolyakrasnik** [comment]
Yep, it's correct behavior.  The faster way to get element is to use class chain strategy. The slowest way - is to use xpath. Speed doesn't depend on carina framework in this case. There is a guide how to use class chain in iOS: https://github.com/facebook/WebDriverAgent/wiki/Class-Chain-Queries-Construction-Rules
To use class chains in carina all you need is to add annotation to element. There is an example:
```
@FindBy(xpath = "**/XCUIElementTypeNavigationBar/XCUIElementTypeOther")
@ClassChain
private ExtendedWebElement screenTitle; 
```
